src/Symfony/Component/Form/Extension/Validator/ViolationMapper/ViolationMapper.php

```diff
@@ -188,7 +186,7 @@ private function matchChild(FormInterface $form, PropertyPathIteratorInterface $
             }
 
             /** @var FormInterface $child */
-            foreach ($children as $key => $child) {
+            foreach ($children as $i => $child) {
```

Comment:
Moreover, `i` implies it's just counter indeed. But also implies it's an integer, IMO.